### PR TITLE
get durable subscription without handling startMessageRollbackDurationSec

### PR DESCRIPTION
**Pull request by linlinnn**
**Motivation**
handle startMessageRollbackDurationSec when get durable subscription

### CODE REVIEW

**Review by eolivelli — CHANGES_REQUESTED** [review]
Nice.

Can you please add a test case?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 1]
```diff
@@ -893,24 +898,29 @@ public void openCursorFailed(ManagedLedgerException exception, Object ctx) {
...
         }
     }
 
+    private void resetSubscriptionCursor(Subscription subscription, CompletableFuture<Subscription> subscriptionFuture,
+                                         long startMessageRollbackDurationSec) {
+        long timestamp = System.currentTimeMillis()
+                - TimeUnit.SECONDS.toMillis(startMessageRollbackDurationSec);
+        final Subscription finalSubscription = subscription;
+        subscription.resetCursor(timestamp).handle((s, ex) -> {
+            if (ex != null) {
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:915]: If the reset cursor operation is failed, the subscription still could be generated successfully, it seems that the configuration startMessageRollbackDurationSec is invalid, this has no effect on the users, right?

Maybe we could print a simple error message in the warning log.

**Comment by linlinnn** [comment]
> Can you please add a test case?
@eolivelli any suggestion about this, `MockManagedLedger` do nothing.

**Comment by linlinnn** [comment]
@eolivelli I have added unit test, please review again, thanks

**Review by eolivelli — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]

**Pull request merged**